src/main/java/frc/robot/subsystems/SwerveSubsystem.java

```diff
@@ -223,10 +249,76 @@ public Command followTrajectory(String name, PathPlannerTrajectory trajectory){
         ).beforeStarting(() -> trajectoryLog.append("Following trajectory " + name)
         ).andThen(() -> trajectoryLog.append("Trajectory " + name +  " Ended"));
     }
-    
+
+    public Command alignToTarget(NodePosition nodePosition){
+        Integer id = Photonvision.getAprilTagID();
+        double xOffset = 0; //Range offset //TODO: Add constants to calculate 
+        double yOffset = 0; //Left/Right offset //TODO: Add TOF sensor method to calculate
+
+        if(id != -1 && Robot.isRed ? id < 4 : id > 5){
+            //Get pose of node to be targeted
+            Pose2d desiredPose = new Pose2d(
+                nodePosition.getTranslation(id),
+                null
+            );
+            //Apply offsets and add a rotation to get desired robot pose
+            desiredPose = new Pose2d(
+                desiredPose.getX() + xOffset, 
+                desiredPose.getY() + yOffset, 
+                Robot.isRed ? Rotation2d.fromDegrees(0) : Rotation2d.fromDegrees(180)
+            );
+            //Generate trajectory to desired pose
+            Pose2d currentPose = getPose();
+            return followTrajectory(
+                "Align",
+                AutonSubsytem.generateTrajectory(
+                    AutonSubsytem.constructPoint(
+                        currentPose, 
+                        Units.radiansToDegrees(
+                            Math.atan2(
+                                desiredPose.getY() - currentPose.getY(), 
+                                desiredPose.getX() - desiredPose.getX()
+                            )
+                        )
+                    ), 
+                    AutonSubsytem.constructPoint(
+                        desiredPose,
+                        currentPose.getY() > desiredPose.getY() ? -90 : 90)
+                )
+            );
```

Comment:
Why can't we use goToPose here? I see the last point heading is different, though I'm unclear why. 

Adding some comments on the intention would be helpful.

Reply:
The last point's heading is different because it creates a better trajectory for scoring
I didn't mean to push goToPose, it's just a method for testing our current pose estimation (checking if it can reliably go to a pose after driving around)